shaders/src/surface_depth_main.fs

```diff
@@ -63,8 +63,8 @@ void main() {
     outPicking.g = mod(float(object_uniforms_objectId)        , 256.0) / 255.0;
     outPicking.r = vertex_position.z / vertex_position.w;
 #else
-    outPicking.x = intBitsToFloat(object_uniforms_objectId);
-    outPicking.y = vertex_position.z / vertex_position.w;
+    outPicking.x = object_uniforms_objectId;
+    outPicking.y = floatBitsToInt(vertex_position.z / vertex_position.w);
 #endif
```

Comment:
A potential workaround is just to add another path for metal.

Reply:
yeah, we could, but I think this is basically a public API issue too.  We're not clear about which readPIxels format are guaranteed to work; but I think using the same ones that GLES3.0 makes sense?